### PR TITLE
Normalise CI

### PR DESCRIPTION
**Pull request by achamayou**

### CODE REVIEW

**Comment by achamayou** [comment]
WIP for #416, still missing:

- Perf tests
- Coverage reporting (needs labels to recognise BFT/CFT coverage)

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

File: tests/coverage/generate_coverage.sh [hunk 1]
```diff
@@ -19,10 +21,12 @@ llvm-cov-7 export -instr-profile coverage.profdata -format=text "${objects[@]}"
...
 # Generate and upload combined coverage report for Codecov
 llvm-cov-7 show -instr-profile coverage.profdata "${objects[@]}" -ignore-filename-regex="(openenclave|3rdparty|/test/)" > codecov.txt
-bash <(curl -s https://codecov.io/bash) -t "${CODECOV_TOKEN}" -f codecov.txt -F unit
+bash <(curl -s https://codecov.io/bash) -t "${CODECOV_TOKEN}" -f codecov.txt -F "unit_${suffix}"
 
 for e2e in *.virtual.so; do
-    llvm-profdata-7 merge -sparse ./*_"$e2e".profraw -o "$e2e".profdata
-    llvm-cov-7 show -instr-profile "$e2e".profdata -object cchost.virtual -object "$e2e" -ignore-filename-regex="(openenclave|3rdparty|/test/)" > "$e2e".txt
-    bash <(curl -s https://codecov.io/bash) -t "${CODECOV_TOKEN}" -f "$e2e".txt -F "$(echo $"e2e" | cut -d. -f1)"
+    if [ -f ./*_"$e2e".profraw ]; then
```
Comment (achamayou) [inline tests/coverage/generate_coverage.sh]: Not all end to end tests run with PBFT yet, the profraw may legitimately not exist.

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by achamayou** [comment]
SAN + Coverage is just too slow for the current tests, which is why we split it. Keeping it split, this should now be complete and is ready to replace existing CI pipelines.

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**